optimum/exporters/tasks.py

```diff
@@ -2068,9 +2062,16 @@ def get_model_from_task(
                 if original_task == "auto" and config.architectures is not None:
                     model_class_name = config.architectures[0]
 
-        model_class = TasksManager.get_model_class_for_task(
-            task, framework, model_type=model_type, model_class_name=model_class_name, library=library_name
-        )
+        if library_name == "diffusers":
+            config = DiffusionPipeline.load_config(model_name_or_path, **kwargs)
+            class_name = config.get("_class_name", None)
+            loaded_library = importlib.import_module(library_name)
+            model_class = getattr(loaded_library, class_name)
+        else:
+            model_class = TasksManager.get_model_class_for_task(
+                task, framework, model_type=model_type, model_class_name=model_class_name, library=library_name
+            )
+
```

Comment:
what about using `get_model_class_for_task` from tasks manager if the task is explicitly set by user, and the `_class_name` path if task is left as `auto`

Reply:
@IlyasMoutawwakil So far `infer_task_from_model` infer the task with `auto_model_class`, since I did not override it just for pix2pix, the inferred task turns to be `text-to-image`, for which we will drop to the diffusers auto pipeline and lead to the previous error that I reported.
And if I override `auto_model_class` to `StableDiffusionInstructPix2PixPipeline`, with the following code
https://github.com/huggingface/optimum/blob/23f8574ee540e7e4b16ed6537098630ba1c404f4/optimum/onnxruntime/modeling_ort.py#L644-L645
The task inference will fail with following error:
```
ValueError: The task name could not be automatically inferred. If using the command-line, please provide the argument --task task-name. Example: `--task text-classification`.
```

Shall we instead of raising the value error with `auto_model_class`, set the task to "auto"? If so we can do what you suggested, otherwise, we will never have task=="auto".

Reply:
It seems that the purpose of `infer_task_from_model` is to never leave the task as "auto", doubt what I said above is a good idea. Actually all diffusion checkpoints all have their specific model class stored in the `model_index.json` with which we can instantiate the pipeline without issue(at least I haven't seen any issue so far), not sure why we go with auto pipelines for loading the model, since it doesn't contain every specific pipeline ...

Reply:
Thanks for investigating this possibility. I see for now let's just merge since the `_model_class` will always be exportable.